cmd/pdcli/alerts/cmd.go

```diff
@@ -0,0 +1,298 @@
+	var status []string
+	var teams []string
+	var users []string
+
+	var opts pagerduty.ListIncidentsOptions
+
+	var aerr pagerduty.APIError
+
+	// Get current user details
+	user, err := client.GetCurrentUser(pagerduty.GetCurrentUserOptions{})
+
+	if err != nil {
+		if errors.As(err, &aerr) {
+			if aerr.RateLimited() {
+				fmt.Println("rate limited")
+				return nil, err
+			}
+
+			fmt.Println("status code:", aerr.StatusCode)
+
+			return nil, err
+		}
+	}
+
+	// Check the assigned-to flag
+	switch args.assignment {
+
+	case "team":
+		// Fetch incidents belonging to a specific team
+		opts.TeamIDs = append(teams, constants.TeamID)
+
+	case "silentTest":
+		// Fetch incidents assigned to silent test
+		opts.UserIDs = append(users, constants.SilentTest)
+
+	case "self":
+		// Fetch incidents only assigned to self
+		opts.UserIDs = append(users, user.ID)
+	}
+
+	// Fetch only triggered, acknowledged incidents (not resolved ones)
+	opts.Statuses = append(status, constants.StatusTriggered, constants.StatusAcknowledged)
+
+	// Let the number of incidents fetched
+	opts.Limit = constants.AlertsLimit
+
+	// Check urgency
+	if args.low {
+		opts.Urgencies = []string{"low"}
+	} else if args.high {
+		opts.Urgencies = []string{"high"}
+	}
+
+	incidents, err := client.ListIncidentsWithContext(context.Background(), opts)
+
+	if err != nil {
+		if errors.As(err, &aerr) {
+			if aerr.RateLimited() {
+				fmt.Println("rate limited")
+				return nil, err
+			}
+
+			fmt.Println("status code:", aerr.StatusCode)
+
+			return nil, err
+		}
+	}
+
+	// Check if there are no incidents returned
+	if len(incidents.Incidents) == 0 {
+		fmt.Println("Currently there are no alerts assigned to " + args.assignment)
+		os.Exit(1)
+	}
+
+	for _, incident := range incidents.Incidents {
+		incidentIDs = append(incidentIDs, incident.Id)
+	}
+
+	return incidentIDs, nil
+}
+
+// getIncidentAlerts returns all the alerts belong to a particular incident.
+func getIncidentAlerts(client *pagerduty.Client, incidentID string) ([]alert, error) {
+
+	var alerts []alert
+
+	var opts pagerduty.ListIncidentAlertsOptions
+
+	incidentAlerts, err := client.ListIncidentAlertsWithContext(context.Background(), incidentID, opts)
+
+	if err != nil {
+		var aerr pagerduty.APIError
+
+		if errors.As(err, &aerr) {
+			if aerr.RateLimited() {
+				fmt.Println("rate limited")
+				return nil, err
+			}
+
+			fmt.Println("status code:", aerr.StatusCode)
+
+			return nil, err
+		}
+
+	}
+
+	for _, p := range incidentAlerts.Alerts {
+
+		tempAlertObj := alert{}
+
+		tempAlertObj.incidentID = incidentID
+		tempAlertObj.name = p.Summary
+		tempAlertObj.severity = p.Severity
+		tempAlertObj.status = p.Status
+		tempAlertObj.clusterID = fmt.Sprint(p.Body["details"].(map[string]interface{})["cluster_id"])
+
+		alerts = append(alerts, tempAlertObj)
+	}
+
+	return alerts, nil
+}
+
+// printAlerts prints all the alerts to the console in a tabular format.
+func printAlerts(alerts []alert) {
```

Comment:
Please check default printed table columns:
on `$ ./pdcli alerts --assigned-to team`
```
  CLUSTER ID |                  STATUS                   |               SEVERITY               | INCIDENT ID |   NAME    
-------------+-------------------------------------------+--------------------------------------+-------------+-----------
  PZAM54B    | api-ErrorBudgetBurn CRITICAL              | 9d4a1914-8185-4949-8658-4501f022e50e | triggered   | critical  
             | (2)                                       |                                      |             |           
  PZAM54B    | ClusterOperatorDown CRITICAL              | 9d4a1914-8185-4949-8658-4501f022e50e | triggered   | critical  

```

```
  CLUSTER ID |                  STATUS                   |               SEVERITY               | INCIDENT ID |   NAME    
-------------+-------------------------------------------+--------------------------------------+-------------+-----------
  PZAM54B    | api-ErrorBudgetBurn CRITICAL              | 9d4a1914-8185-4949-8658-4501f022e50e | triggered   | critical  
             | (2)                                       |                                      |             |           

```

Reply:
There was an issue with the map keys. They were being appended to the slice not in-order. Fixed it.